src/api.js

```diff
@@ -1651,14 +1651,17 @@ api.declare({
 
   // Create temporary credentials for the task
   let credentials = taskcluster.createTemporaryCredentials({
-    start:  new Date(Date.now() - 15 * 60 * 1000),
-    expiry: new Date(takenUntil.getTime() + 15 * 60 * 1000),
+    start:  new Date(),
+    expiry: takenUntil,
     scopes: [
       'queue:reclaim-task:' + taskId + '/' + runId,
       'queue:resolve-task:' + taskId + '/' + runId,
       'queue:create-artifact:' + taskId + '/' + runId,
     ].concat(task.scopes),
     credentials: this.credentials,
+    clientId:
+      'task-client/' + '/' + runId + '/on/' + run.workerGroup + '/' +
+      run.workerId + '/until/' + takenUntil.getTime(),
```

Comment:
I propose `task/${taskId}/${runId}@${workerGroup}/${workerId}`, the takenUntil is nice in theory, but will we ever need it in practice...